### PR TITLE
bump manifest-tool to latest

### PR DESCRIPTION
**Pull request by deitch**
Signed-off-by: Avi Deitcher <avi@deitcher.net>

**- What I did**

Updated `manifest-tool` to the latest

**- How I did it**

With much `vndr` pain

**- How to verify it**

Run it. I ran and pushed some images and signed them

**- Description for the changelog**

Latest manifest-tool library

**Additional information**

Some important notes:

First, this was necessary because of the work for dockerless, which was blocked by a version incompatibility with an earlier version of manifest-tool

Second, there is a new file `src/cmd/linuxkit/pkglib/manifest.go`.

This entire file - all 300 lines - is because one of the library utilities we used in manifest-tool went private and started to depend on CLI. I expect it eventually will go away, and we will be able to delete this file. I already reached out to Phil about it.

This means... please don't comment on the goodness or badness or commentness or anything else about that file, unless it is something egregious. I hope to make it all disappear in the next few weeks.

### CODE REVIEW

**Comment by deitch** [comment]
Oh good! CI is clean.

File: src/cmd/linuxkit/moby/docker.go [hunk 1]
```diff
@@ -19,6 +19,7 @@ import (
 	"github.com/docker/cli/cli/config/configfile"
 	distref "github.com/docker/distribution/reference"
 	"github.com/docker/docker/api/types"
+	apitypes "github.com/docker/docker/api/types"
```
Comment (justincormack) [inline src/cmd/linuxkit/moby/docker.go:22]: you are including this twice with different names!

Comment (justincormack) [reply src/cmd/linuxkit/moby/docker.go:22]: elsewhere you include it as dockertypes...

File: src/cmd/linuxkit/pkglib/manifest.go [hunk 2]
```diff
@@ -0,0 +1,295 @@
...
+	"github.com/estesp/manifest-tool/pkg/store"
+	"github.com/estesp/manifest-tool/pkg/types"
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	log "github.com/sirupsen/logrus"
+)
+
+/*
+ EVERYTHING below here is because github.com/estesp/manifest-tool moved pushManifestList into
+ a non-exported func and passed it cli. If/when it moves back, we can get rid of all of it.
+*/
```
Comment (justincormack) [inline src/cmd/linuxkit/pkglib/manifest.go:31]: if this code is derived from that code you should mention it!

File: src/cmd/linuxkit/moby/docker.go (hunk 1 above)
Comment (deitch) [reply src/cmd/linuxkit/moby/docker.go:22]: OK, standardized on `dockertypes`

File: src/cmd/linuxkit/pkglib/manifest.go (hunk 2 above)
Comment (deitch) [reply src/cmd/linuxkit/pkglib/manifest.go:31]: Done

**Comment by deitch** [comment]
Updated per your two comments @justincormack

File: src/cmd/linuxkit/moby/docker.go [hunk 3]
```diff
@@ -19,6 +19,7 @@ import (
 	"github.com/docker/cli/cli/config/configfile"
 	distref "github.com/docker/distribution/reference"
 	"github.com/docker/docker/api/types"
+	dockertypes "github.com/docker/docker/api/types"
```
Comment (justincormack) [inline src/cmd/linuxkit/moby/docker.go:22]: you have still included it twice, it is the line above.

Comment (deitch) [reply src/cmd/linuxkit/moby/docker.go:22]: Oh now I see it. OK thanks. Fixed.

**Comment by deitch** [comment]
Updated to fix that @justincormack

**Comment by deitch** [comment]
Rebased this based upon master with the changes @justincormack put in for darwin-arm64 in #3577 , keeping that latest version of containerd. Looking good. @justincormack give another review?

**Review by justincormack — APPROVED** [review]

**Pull request merged**